link_shorterner/src/shortener_link_handler.erl

```diff
@@ -0,0 +1,33 @@
+-module(shortener_link_handler).
+
+-export([init/2]).
+
+init(Req, State) ->
+    Url = get_request_url(Req),
+    ReqMethod = cowboy_req:method(Req),
+    {Status, ShortUrl} = handle_request(ReqMethod, Url),
+    Body = "{url: \"" ++ ShortUrl ++ "\"}",
+    Header = #{<<"content-type">> => <<"application/json">>},
+    Resp = cowboy_req:reply(Status, Header, Body, Req),
+    {ok, Resp, State}.
+
+handle_request(<<"POST">>, Url) -> 
+    {CreationStatus, ShortUrl} = shortener_shortener:short(Url),
+    HttpStatus =
+        case CreationStatus of
+            old -> 200; %ok
+            new -> 201  %created
+        end,
+    {HttpStatus, ShortUrl};
+    
+handle_request(<<"GET">>, Url) ->
+    case shortener_shortener:get(Url) of
+        not_found -> {404, ""};
+        ShortUrl  -> {302, ShortUrl}
+    end.
+
+get_request_url(Req) ->
+    <<"/",UrlBinary/binary>> = cowboy_req:path(Req),
+    binary_to_list(UrlBinary).
```

Comment:
try to stick to binaries unless there's a really good reason not to. Binary is the string type in Erlang.

If you need to concatenate you can do stuff like `<<Bin1/binary, "hello", Bin2/binary>>` or, in more complex scenarios, use `iolist_to_binary`. There are a couple of reads on the topic in trello

Reply:
+1